contribs/nocycles/main.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm"
+	"github.com/gnolang/gno/gnovm/pkg/gnoenv"
+	"github.com/gnolang/gno/gnovm/pkg/gnolang"
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/gnovm/pkg/packages"
+)
+
+func main() {
+	// set to true to dump the final pkg list
+	verbose := true
+
+	// find stdlibs
+	libs := []string{}
+	gnoRoot := gnoenv.RootDir()
+	stdlibsDir := filepath.Join(gnoRoot, "gnovm", "stdlibs")
```

Comment:
for stdlibs, instead, i think you could change `misc/genstd` (note: this binary as well, if it were to stay, would also have to be misc/), by having its import order generator also consider test files

Reply:
Indeed, when I cherry-pick d9f4d3af653043df177091b916badae6adeb2896 on master, I get:
> go run github.com/gnolang/gno/misc/genstd
panic: cyclical package initialization on "bufio" (bufio -> bufio)

I like having code that is dedicated to detecting any cycles in stdlibs and examples though, I'll move the `nocycles`code into a test somewhere I think when I get to the polishing step on this PR

Reply:
Indeed, when I cherry-pick 4dbc66a22ca66260f2f1418f6382f667c6c23893 on master, I get:
> go run github.com/gnolang/gno/misc/genstd
panic: cyclical package initialization on "bytes" (bytes -> io -> bytes)

I like having code that is dedicated to detecting any cycles in stdlibs and examples though, I'll move the `nocycles` code into a test somewhere I think when I get to the polishing step on this PR

Reply:
moved to `examples/no_cycles_test.go`

Reply:
regarding genstd changes:

actually finding an global order while including all test files is an undecidable problem, consider the following imports (which are allowed in go):
- `foopkg`/foo_test.go -> `barpkg`
- `barpkg`/bar_test.go -> `foopkg`

examining each packages tests deps will yield a different order:
- `foopkg` tests:
  - `barpkg`
  - `foopkg`
- `barpkg` tests:
  - `foopkg`
  - `barpkg`
  
we could create a `map[string][]string` to get the testing order for each lib but I'm not sure this is really helpful